code/modules/hallucinations/types/powers.dm

```diff
@@ -0,0 +1,165 @@
+	return ..()
+
+/datum/hallucination/mindread/activate()
+	..()
+	addtimer(CALLBACK(src, .proc/mind_give), rand(30, 50))
+
+/datum/hallucination/mindread/start()	//set duration, foreshadow powers
+	duration = rand(2, 4) MINUTES
+	switch(rand(1, 3))
+		if(1)
+			sound_to(holder, 'sound/misc/announcements/notice.ogg')
+			to_chat(holder, "<h2 class='alert'>Ion Storm?</h2>")
+			to_chat(holder, SPAN_ALERT("It has come to our attention that the station has passed through an unusual ion storm. Several crewmembers are exhibiting unusual abilities."))
+		if(2)
+			sound_to(holder, 'sound/hallucinations/behind_you1.ogg')
+			to_chat(holder, SPAN_GOOD("You hear a whispering in your mind. A promise of [pick("power", "enlightenment", "sight beyond sight", "knowledge terrible but true")]. Your vision goes white for a moment; when it returns, you feel... different."))
+			flick("e_flash", holder.flash)
+		if(3)
+			to_chat(holder, FONT_LARGE(SPAN_DANGER("You feel a sudden pain in your head, as if it's being ripped in two! When it subsides to a dull throbbing a moment later, you feel... different.")))
+			holder.emote("me",1,"winces.")
+			holder.eye_blurry += 9
+
+/datum/hallucination/mindread/proc/mind_give()	//grant powers
+	to_chat(holder, SPAN_NOTICE(FONT_LARGE("<B>You have developed a psionic gift!</B>")))
+	to_chat(holder, SPAN_NOTICE("You can feel your mind surging with power! Check the abilities tab to use your new power!"))
+	holder.verbs += /mob/living/carbon/proc/fakemindread
+
+/datum/hallucination/mindread/end()
+	if(holder)
+		holder.verbs -= /mob/living/carbon/proc/fakemindread
+		to_chat(holder, SPAN_NOTICE("<b>Your psionic powers vanish abruptly, leaving you cold and empty.</b>"))
+		holder.drowsyness += 12
+	..()
+
+/mob/living/carbon/proc/fakemindread()
+	set name = "Read Mind"
+	set category = "Abilities"
+
+	if(!hallucination)
+		for(var/datum/hallucination/mindread/H in hallucinations)
+			H.end()
+		return
+
+	if(stat)
+		to_chat(usr, SPAN_WARNING("You're not in any state to use your powers right now!"))
+		return
+
+	var/list/creatures = list()
+	for(var/mob/living/C in oview(usr))
+		creatures += C
+	creatures -= usr
+	if(!creatures.len)
+		return
+
+	var/mob/target = input("Whose mind do you wish to probe?") as null|anything in creatures
+	if(isnull(target))
+		return
+	if(target.stat)
+		to_chat(usr, SPAN_WARNING("\The [target]'s mind is not in any state to be probed!"))
+		return
+
+	to_chat(usr, SPAN_NOTICE("<b>You dip your mentality into the surface layer of \the [target]'s mind, seeking a prominent thought.</b>"))
+	if(do_after(usr, 30))
+		sleep(rand(50, 120))
+		to_chat(usr, SPAN_NOTICE("<b>You skim thoughts from the surface of \the [target]'s mind: \"<i>[pick(SShallucinations.hallucinated_phrases)]</i>\"</b>"))
+		for(var/mob/living/carbon/human/M in oviewers(src))
+			to_chat(M, "<B>[usr]</B> puts [usr.get_pronoun(1)] hands to [usr.get_pronoun(1)] head and mumbles incoherently as they stare, unblinking, at \the [target].")
```

Comment:
I am pretty sure these can be made into a single visible_message? 
`/mob/visible_message(var/message, var/self_message, ...)`
There's also more occurrances of this around, so fix those too.